src/main/kotlin/com/stefan_grafberger/streamdq/checks/row/ListLengthInRange.kt

```diff
@@ -0,0 +1,56 @@
+package com.stefan_grafberger.streamdq.checks.row
+
+import com.stefan_grafberger.streamdq.checks.RowLevelConstraintResult
+import com.stefan_grafberger.streamdq.checks.TypeQueryableRowMapFunction
+import org.apache.commons.math3.stat.inference.TestUtils
+import org.apache.flink.api.common.ExecutionConfig
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.streaming.util.typeutils.FieldAccessor
+import org.apache.flink.streaming.util.typeutils.NullCheckingFieldAccessorFactory
+import java.util.Objects
+import kotlin.reflect.KClass
+
+data class ListLengthInRange(
+    override val keyExpressionString: String,
+    val expectedLowerBound: Int? = null,
+    val expectedUpperBound: Int? = null
+) :
+    RowLevelConstraint() {
+    override fun <T> getCheckResultMapper(
+        streamObjectTypeInfo: TypeInformation<T>,
+        config: ExecutionConfig?
+    ): TypeQueryableRowMapFunction<T> {
+        return ListLengthInRangeRowMapFunction(
+            streamObjectTypeInfo, expectedLowerBound, expectedUpperBound,
+            toString(), keyExpressionString, config
+        )
+    }
+}
+
+class ListLengthInRangeRowMapFunction<T>(
+    streamObjectTypeInfo: TypeInformation<T>,
+    private val expectedLowerBound: Int?,
+    private val expectedUpperBound: Int?,
+    val checkName: String,
+    keyExpressionString: String,
+    val config: ExecutionConfig?
+) : TypeQueryableRowMapFunction<T>() {
+
+    private val fieldAccessor: FieldAccessor<T, Any?>
+    init {
+        fieldAccessor = NullCheckingFieldAccessorFactory.getAccessor(streamObjectTypeInfo, keyExpressionString, config)
+    }
+    override fun map(value: T): RowLevelConstraintResult {
+        // TODO: Support different comparable types
```

Comment:
What other types would we expect here for this check? Are there other List-like classes we should support here? Do they have a shared parent class? If not, we could think about removing this todo

Reply:
Emm yes there might be Set also, hence I changed it to Collection<*> instead of List<*>